fvtest/compilertriltest/TernaryTest.cpp

```diff
@@ -0,0 +1,490 @@
+ *******************************************************************************/
+
+#include "OpCodeTest.hpp"
+#include "default_compiler.hpp"
+#include "omrformatconsts.h"
+
+// C++11 upgrade (Issue #1916).
+template <typename CompareType, typename ValType>
+class TernaryTest : public TRTest::JitTest, public ::testing::WithParamInterface<std::tuple<std::tuple<CompareType, CompareType>, std::tuple<ValType, ValType>, ValType (*)(CompareType, CompareType, ValType, ValType)>> {};
+
+/** \brief
+ *     Struct equivalent of the TernaryTestParamStruct tuple.
+ *
+ *  \tparam CompareType
+ *     The type of the compare child of the ternary opcode.
+ *
+ *  \tparam ValType
+ *     The type that the ternary opcode returns.
+ */
+template <typename CompareType, typename ValType>
+struct TernaryTestParamStruct
+   {
+   CompareType c1;
+   CompareType c2;
+   ValType v1;
+   ValType v2;
+   ValType (*oracle)(CompareType, CompareType, ValType, ValType);
+   };
+
+/** \brief
+ *     Given an instance of TernaryTestParamStruct, returns an equivalent instance of TernaryTestParamStruct.
+ */
+template <typename CompareType, typename ValType>
+TernaryTestParamStruct<CompareType, ValType> to_struct(std::tuple<std::tuple<CompareType, CompareType>, std::tuple<ValType, ValType>, ValType (*)(CompareType, CompareType, ValType, ValType)> param)
+   {
+   TernaryTestParamStruct<CompareType, ValType> s;
+
+   s.c1 = std::get<0>(std::get<0>(param));
+   s.c2 = std::get<1>(std::get<0>(param));
+   s.v1 = std::get<0>(std::get<1>(param));
+   s.v2 = std::get<1>(std::get<1>(param));
+   s.oracle = std::get<2>(param);
+   return s;
+   }
+
+/** \brief
+ *     The oracle function which computes the expected result of the ternary test.
+ *
+ *  \tparam CompareType
+ *     The type of the values compared by oracle function.
+ *  \tparam ValType
+ *     The return type of the oracle function.
+ */
+template <typename CompareType, typename ValType>
+static ValType xternaryOracle(CompareType c1, CompareType c2, ValType v1, ValType v2)
+   {
+   return ((c1 < c2) ? v1 : v2);
+   }
+
+/** \brief
+ *     The function computes the input vector of pairs for the compare node.
+ */
+template <typename T>
+static std::vector<std::pair<T, T>> compareInputs()
+   {
+   std::pair<T, T> inputArray[] =
+      {
+      std::make_pair(std::numeric_limits<T>::min(), std::numeric_limits<T>::max()),
+      std::make_pair(std::numeric_limits<T>::max(), std::numeric_limits<T>::min()),
+      std::make_pair(std::numeric_limits<T>::min(), std::numeric_limits<T>::min()),
+      std::make_pair(0, std::numeric_limits<T>::max()),
+      std::make_pair(0, std::numeric_limits<T>::min()),
+      std::make_pair(std::numeric_limits<T>::min(), 0),
+      std::make_pair(std::numeric_limits<T>::max(), 0),
+      };
+   return std::vector<std::pair<T, T>>(inputArray, intputArray + sizeof(intputArray)/sizeof(std::pair<T, T>));
```

Comment:
@yanluo7 A few typos apparently:

```suggestion
   return std::vector<std::pair<T, T>>(inputArray, inputArray + sizeof(inputArray)/sizeof(std::pair<T, T>));
```